lib/app.js

```diff
@@ -33,110 +33,115 @@ function app (opts) {
     request(options, opts.throttle)
       .then(scriptData.parse)
     // comment next line to get raw data
-      .then(scriptData.extractor(MAPPINGS))
+      .then(scriptData.extractor(MAPPINGS, 'Ws7gDc'))
```

Comment:
I checked a few apps and it looks like two sections have the same id. I'm not sure if the section that contains the details is the first one is a coincidence, or it will work for all cases. Just wondering if there could be another way to find the correct section. Maybe check the appId path for all sections and match the one that matches the input?

Reply:
```
        "ds:5": {
            "id": "Ws7gDc",
        },
        "ds:6": {
            "id": "Ws7gDc"
```

Reply:
I'm fine with the appId approach. I'm kinda busy right now so it might take some time for me to rewrite it.

Reply:
@knirpsfass I just wrote it, would you please try it and see if it works better? All changes are in one file only.
I'm not sure what the familyGenre is, I would just leave it without changing.
```
    request(options, opts.throttle)
      .then(scriptData.parse)
    // comment next line to get raw data
      .then(appObject => getDetailsCluster(appObject, opts.appId))
      .then(scriptData.extractor(MAPPINGS))
      .then(R.assoc('url', reqUrl))
      .then(resolve)
      .catch(reject);
  });
}

function getDetailsCluster(appObject, appId) {
  for (let cluster in appObject) {
    if (R.path(MAPPINGS.appId, appObject[cluster]) === appId) {
      return appObject[cluster];
    }
  }
  // this should never run:
  return appObject["ds:4"];
}

const MAPPINGS = {
  appId: [1, 2, 77, 0],
  title: [1, 2, 0, 0],

```

Reply:
Or you can do it without creating a new fuction:

```
request(options, opts.throttle)
      .then(scriptData.parse)
      .then(R.values)
      .then(clusters => clusters.filter(cluster => R.path(MAPPINGS.appId, cluster) === opts.appId)[0])
      .then(scriptData.extractor(MAPPINGS))
      .then(R.assoc('url', reqUrl))
      .then(resolve)
      .catch(reject);
```

Reply:
One issue with this is that it will miss information in different sections. Like `comments` I think there should be separate mappings object for such fields.